nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/excel/ExcelHeaderSchemaStrategy.java

```diff
@@ -0,0 +1,116 @@
+import org.apache.nifi.serialization.record.RecordField;
+import org.apache.nifi.serialization.record.RecordSchema;
+import org.apache.nifi.util.SchemaInferenceUtil;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.DataFormatter;
+import org.apache.poi.ss.usermodel.Row;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.EnumSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class ExcelHeaderSchemaStrategy implements SchemaAccessStrategy {
+    private static final Set<SchemaField> schemaFields = EnumSet.noneOf(SchemaField.class);
+
+    private final PropertyContext context;
+
+    private final DataFormatter dataFormatter;
+
+    private final ComponentLog logger;
+
+    public ExcelHeaderSchemaStrategy(PropertyContext context, ComponentLog logger) {
+        this(context, logger, null);
+    }
+
+    public ExcelHeaderSchemaStrategy(PropertyContext context, ComponentLog logger, Locale locale) {
+        this.context = context;
+        this.logger = logger;
+        this.dataFormatter = locale == null ? new DataFormatter() : new DataFormatter(locale);
+    }
+
+    @Override
+    public RecordSchema getSchema(Map<String, String> variables, InputStream contentStream, RecordSchema readSchema) throws SchemaNotFoundException, IOException {
+        if (this.context == null) {
+            throw new SchemaNotFoundException("Schema Access Strategy intended only for validation purposes and cannot obtain schema");
+        }
+
+        String errMsg = "Failed to read Header line from Excel worksheet";
+        RecordSource<Row> recordSource;
+        try {
+            recordSource = new ExcelRecordSource(contentStream, context, variables, logger);
+        } catch (Exception e) {
+            throw new SchemaNotFoundException(errMsg, e);
+        }
+
+        Row headerRow = recordSource.next();
+        if (!ExcelUtils.hasCells(headerRow)) {
+            throw new SchemaNotFoundException("The chosen header line in the Excel worksheet had no cells");
+        }
+
+        try {
+            String dateFormat = context.getProperty(DateTimeUtils.DATE_FORMAT).getValue();
+            String timeFormat = context.getProperty(DateTimeUtils.TIME_FORMAT).getValue();
+            String timestampFormat = context.getProperty(DateTimeUtils.TIMESTAMP_FORMAT).getValue();
+            final TimeValueInference timeValueInference = new TimeValueInference(dateFormat, timeFormat, timestampFormat);
+            final Map<String, FieldTypeInference> typeMap = new LinkedHashMap<>();
+            IntStream.range(0, headerRow.getLastCellNum())
+                    .forEach(index -> {
+                        final Cell cell = headerRow.getCell(index);
+                        final String fieldName = Integer.toString(index);
```

Comment:
@exceptionfactory Actually that is the expected behavior as this is not CSV hence the values in the header column do not necessarily have to be strings. I used the field index since it is predictable and unchanging. Personally I am really not sure why this strategy is necessary once there is the inference strategy which is more encompassing than the header strategy..

Reply:
@exceptionfactory Also the other problem  or "feature" with that is that all the field types will be strings.
I am open to whatever you prefer. Let me know whether we should remove it or modify it.

Reply:
The string typing is also a problem as you noted. Theoretically that could also be changed to follow an approach similar to the Infer Schema strategy, reading all rows to determine potential values. In the interest of completing a first version, this strategy could be removed for now and evaluated more closely in a follow-on Jira issue.

Reply:
Done